docs/proposals/image-provenance.md

```diff
@@ -0,0 +1,329 @@
+- For non-replicated things (size 1 ReplicaSet, PetSet), a single node failure may disable it.
+- a node rolling update will eventually check for liveness of replacements, and would be throttled if
+  in the case when the image was no longer allowed and so replacements could not be started.
+- rapid node restarts will cause existing pod objects to be restarted by kubelet.
+- slow node restarts or network partitions will cause node controller to delete pods and there will be no replacement
+
+It is up to the Backend implementor, and the cluster administrator who decides to use that backend, to decide
+whether the Backend should be allowed to change its mind.  There is a tradeoff between responsiveness
+to changes in policy, versus keeping existing services running.  The two models that make sense are:
+
+- never change a policy, unless some external process has ensured no active objects depend on the to-be-forbidden
+  images.
+- change a policy and assume that transition to new image happens faster than the existing pods decay.
+
+## Ubernetes
+
+If two clusters share an image policy backend, then they will have the same policies.
+
+The clusters can pass different tokens to the backend, and the backend can use this to distinguish
+between different clusters.
+
+## Image tags and IDs
+
+Image tags are like: `myrepo/myimage:v1`.
+
+Image IDs are like: `myrepo/myimage@sha256:beb6bd6a68f114c1dc2ea4b28db81bdf91de202a9014972bec5e4d9171d90ed`.
+You can see image IDs with `docker images --no-trunc`.
+
+The Backend needs to be able to resolve tags to IDs (by talking to the images repo).
+If the Backend resolves tags to IDs, there is some risk that the tag-to-ID mapping will be
+modified after approval by the Backend, but before Kubelet pulls the image.  We will not address this
+race condition at this time.
+
+We will wait and see how much demand there is for closing this hole. If the community demands a solution,
+we may suggest one of these:
+
+1.  Use a backend that refuses to accept images that are specified with tags, and require users to resolve to IDs
+    prior to creating a pod template.
+   - [kubectl could be modified to automate this process](https://github.com/kubernetes/kubernetes/issues/1697)
+   - a CI/CD system or templatingcould be used that maps IDs to tags before Deployment modification/creation.
+1. Audit logs from kubelets to see image IDs were actually run, to see if any unapproved images slipped through.
+1. Monitor tag changes in image repository for suspicious activity, or restrict remapping of tags after initial application.
+
+If none of these works well, we could do the following:
+
+- Image Policy Admission Controller adds new field to Pod, e.g. `pod.spec.container[i].imageID` (or an annotation).
+  and kubelet will enforce that both the imageID and image match the image pulled.
+
+Since this adds complexity and interacts with imagePullPolicy, we avoid adding the above feature initially.
```

Comment:
This definitely deserves a separate issue. I've started wondering if it would make sense to implement here OpenShift's ImageStreams but in a limited way, iow. in such a way that you could have a one-to-one mapping image to a specific image:tag which would then allow a user to specify if I care about this being a hard reference to a particular imageid at the point when that ImageStream got created or rather a following tag, iow. such that would be refreshed on every usage, which would mean it might be different each time. @smarterclayton wdyt?